tasks/libs/ciproviders/github_api.py

```diff
@@ -519,14 +519,29 @@ def get_token_from_app(app_id_env='GITHUB_APP_ID', pkey_env='GITHUB_KEY_B64'):
         auth_token = integration.get_access_token(install_id)
         print(auth_token.token)
 
-    def create_label(self, name, color, description=""):
+    def create_label(self, name, color, description="", ignore_existing=False):
         """
         Creates a label in the given GitHub repository.
         """
-        return self._repository.create_label(name, color, description)
+        import github
 
-    def create_milestone(self, title):
-        self._repository.create_milestone(title)
+        try:
+            return self._repository.create_label(name, color, description)
+        except github.GithubException as e:
+            if not (e.status == 422 and ignore_existing):
```

Comment:
Going by [the docs](https://docs.github.com/en/rest/issues/labels?apiVersion=2022-11-28#create-a-label), just checking the status code wouldn't be enough to assert that the validation failure was due to the label already existing:
>  422 | Validation failed, or the endpoint has been spammed.

I think that when I hit this there was some `reason` field or similar, but I don't see it documented. So it's either that or accepting that this is good enough, or using an extra query to see if the label already exists (before or after).

Reply:
Addressed!